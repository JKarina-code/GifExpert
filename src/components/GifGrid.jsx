export const GifGrid = (category) => {

return (
    <>
     <h3>Your gif</h3>
    
    </>
)



}