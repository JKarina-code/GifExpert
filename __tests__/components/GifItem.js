import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";
describe("test in <GifItem/>", () => {
  const title = "Pokemon";
  const url ="https://user-images.githubusercontent.com/29663094/234428463-bc23cfcb-c5fb-4b77-82b4-e56e6f8fa9fc.png";

  test("must do match with snapshot", () => {
    const { container } = render(<GifItem title={title} src={url} />);

    expect(container).toMatchSnapshot();
  });

  test("must show the image with src and alt", () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("must  show the title", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
