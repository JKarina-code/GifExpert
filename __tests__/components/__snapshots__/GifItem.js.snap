// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifItem/> must do match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pokemon"
    />
    <p>
      Pokemon
    </p>
  </div>
</div>
`;
